.github/workflows/dogfood.yaml

```diff
@@ -72,17 +77,23 @@ jobs:
           tags: "codercom/oss-dogfood:${{ steps.docker-tag-name.outputs.tag }},codercom/oss-dogfood:latest"
 
       - name: Build and push Nix image
-        uses: depot/build-push-action@636daae76684e38c301daa0c5eca1c095b24e780 # v1.14.0
-        with:
-          project: b4q6ltmpzh
-          token: ${{ secrets.DEPOT_TOKEN }}
-          buildx-fallback: true
-          context: "."
-          file: "dogfood/contents/Dockerfile.nix"
-          pull: true
-          save: true
-          push: ${{ github.ref == 'refs/heads/main' }}
-          tags: "codercom/oss-dogfood-nix:${{ steps.docker-tag-name.outputs.tag }},codercom/oss-dogfood-nix:latest"
+        run: |
+          nix build .#dev_image
+
+          # If github.ref is not refs/head/main then exit here early
+           if [ "${{ github.ref }}" != "refs/heads/main" ]; then
+             exit 0
+           fi
+
+          docker load -i result
+
+          CURRENT_SYSTEM=$(nix eval --impure --raw --expr 'builtins.currentSystem')
+
+          docker image tag codercom/oss-dogfood-nix:latest-$CURRENT_SYSTEM codercom/oss-dogfood-nix:${{ steps.docker-tag-name.outputs.tag }}
+          docker image push codercom/oss-dogfood-nix:${{ steps.docker-tag-name.outputs.tag }}
+
+          docker image tag codercom/oss-dogfood-nix:latest-$CURRENT_SYSTEM codercom/oss-dogfood-nix:latest
+          docker image push codercom/oss-dogfood-nix:latest
```

Comment:
Is this multi-arch? The image we were building with depot runners was multi-arch.

Reply:
No, this is a single arch image build.

Also, I can't find us building multi-arch images on Depot. Would you mind pointing me to where we're doing that?

Reply:
I am sorry for the confusion. Multi-arch builds are for our github.com/coder/images repo. Depot does that automatically. 

Dogfood is still `amd64` only.